vendor/src/github.com/docker/libnetwork/resolver_unix.go

```diff
@@ -0,0 +1,77 @@
+// +build !windows
+
+package libnetwork
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"runtime"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/docker/pkg/reexec"
+	"github.com/docker/libnetwork/iptables"
+	"github.com/vishvananda/netns"
+)
+
+func init() {
+	reexec.Register("setup-resolver", reexecSetupResolver)
```

Comment:
@icecrime Yes, it is a reexec but it will only happen when a container is connected to a user-defined network so most container starts like build steps should not see any slowness. Even though the issue this is fixing has been there from 1.10, the fix is extremely critical in order to make sure we don't have OS threads in the daemon in a different namespace.